hyperspy/io.py

```
@@ -478,6 +478,9 @@ def save(filename, signal, overwrite=None, **kwds):
         ensure_directory(filename)
         if overwrite is None:
             overwrite = overwrite_method(filename)
+        if overwrite is False:
```

This "fails" (as in does not overwrite) silently, and also might confuse future developers.

I would prefer this to be formatted along the lines:
```python
if overwrite is False and os.path.isfile(filename):
    raise ValueError("some appropriate text")
# If we get here, either we overwrite or the file does not exist
# Put the code that in your PR starts with line #485 
# (i.e. without the `if overwrite is True`)
```

---

I am a bit confused, are you suggesting replacing lines 481-483 with the above? 

When I test, this PR works correctly for all four combinations: 
1) overwrite = True, file exists --> file is overwritten
2) overwrite = True, file doesn't exist --> new file is written
3) overwrite = False, file exists --> do nothing
4) overwrite = False, file doesn't exist --> new file is written

I am not able to interpret the Travis CI and AppVeyor reports, but if the problem is not doing anything at 3), then raising an error is no better. I was using `save(overwrite=False)` because I wanted no interruptions when looping a process over many files creating multiple output files for each, not knowing whether I had made one of output files before. 

Better would be (starting at line 481): 
```
if overwrite is False:
    if os.path.isfile(filename):
        pass
    else: 
        overwrite = True
```

---

ah I understand what you mean, then: 
´´´
if overwrite is False and os.path.isfile(filename):
   pass
else:
   write file... (line 485 onwards) 

´´´

---

Travis and appveyor are failing because of other tests, which are now fixed in the upstream branch. You can merge upstream/RELEASE_next_patch into your branch if you want to get travis and appveyor passing, but you don't need to worry about this for now!

---

@hentr yes, essentially, but not just `pass`. At the very least you should `_logger.warning(f"file {filename} exists, did not overwrite")`, or similar/better. We try to avoid silently passing things.

---

bah, was still not thorough enough, as inputting `overwrite='whatever'` was the same as `overwrite=True`, so now starting at line 479: 
```python
        if not(overwrite is True or overwrite is False):
            overwrite = overwrite_method(filename)
        if overwrite is False and os.path.isfile(filename):
            _logger.warning(f"file {filename} exists, did not overwrite")
        else:
            writer.file_writer(filename, signal, **kwds)
            ...
```